Quadriga/src/main/java/edu/asu/spring/quadriga/service/impl/passthroughproject/PassThroughProjectManager.java

```diff
@@ -0,0 +1,136 @@
+package edu.asu.spring.quadriga.service.impl.passthroughproject;
+
+import java.util.List;
+import java.util.Properties;
+
+import javax.annotation.Resource;
+import javax.xml.bind.JAXBException;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.asu.spring.quadriga.accesschecks.IProjectSecurityChecker;
+import edu.asu.spring.quadriga.dao.IUserDAO;
+import edu.asu.spring.quadriga.dao.workbench.passthroughproject.IPassThroughProjectDAO;
+import edu.asu.spring.quadriga.dao.workspace.IWorkspaceDAO;
+import edu.asu.spring.quadriga.domain.IUser;
+import edu.asu.spring.quadriga.domain.impl.passthroughproject.PassThroughProjectInfo;
+import edu.asu.spring.quadriga.domain.passthroughproject.IPassThroughProject;
+import edu.asu.spring.quadriga.dto.PassThroughProjectDTO;
+import edu.asu.spring.quadriga.dto.QuadrigaUserDTO;
+import edu.asu.spring.quadriga.exceptions.NoSuchRoleException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaAccessException;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.mapper.PassThroughProjectDTOMapper;
+import edu.asu.spring.quadriga.service.impl.BaseManager;
+import edu.asu.spring.quadriga.service.passthroughproject.IPassThroughProjectManager;
+import edu.asu.spring.quadriga.service.workspace.IExternalWSManager;
+import edu.asu.spring.quadriga.web.login.RoleNames;
+
+/**
+ * 
+ * This class has all the Pass through project service layer functions. It
+ * includes the handling DB and controller services.
+ * 
+ *
+ */
+@Service
+public class PassThroughProjectManager extends BaseManager implements IPassThroughProjectManager {
+
+    @Autowired
+    private IWorkspaceDAO workspaceDao;
+
+    @Autowired
+    private IExternalWSManager externalWSManager;
+
+    @Autowired
+    private IPassThroughProjectDAO projectDao;
+
+    @Autowired
+    private IProjectSecurityChecker projectSecurityChecker;
+
+    @Autowired
+    private IUserDAO userDAO;
+
+    @Autowired
+    private PassThroughProjectDTOMapper projectMapper;
+
+    @Resource(name = "projectconstants")
+    private Properties messages;
+
+    private String addPassThroughProject(IUser user, IPassThroughProject project) throws QuadrigaStorageException {
+        String projectId = projectDao.generateUniqueID();
+        QuadrigaUserDTO owner = userDAO.getDTO(user.getUserName());
+        PassThroughProjectDTO projectDTO = projectMapper.getPassThroughProjectDTO(project, user);
+        projectDTO.setProjectid(projectId);
+        projectDTO.setProjectowner(owner);
+
+        projectDao.addPassThroughProject(projectDTO);
+
+        return projectId;
+    }
+
+    private String getInternalProjectId(String externalProjectid, String userid)
```

Comment:
I think its because even the collaborator admin's credentials can also be used to call the passthrough project api. Please correct me if I am wrong.